src/NServiceBus.Persistence.ServiceFabric/Sagas/SagaInfoCache.cs

```diff
@@ -45,7 +45,13 @@ public void Initialize(SagaMetadataCollection metadataCollection)
         {
             foreach (var metadata in metadataCollection)
             {
-                serializerCache.TryAdd(metadata.SagaEntityType, BuildSagaInfo(metadata.SagaEntityType, metadata.SagaType));
+                RuntimeSagaInfo existing;
+                var sagaDataType = metadata.SagaEntityType;
+                if (serializerCache.TryGetValue(sagaDataType, out existing))
```

Comment:
`Initialize(collection)` is only called from the [`SagaPersistenceFeature.Setup()`](https://github.com/Particular/NServiceBus.Persistence.ServiceFabric/blob/f56aa1d01a981287577748cb348ebf2e63aadb0e/src/NServiceBus.Persistence.ServiceFabric/Sagas/SagaPersistenceFeature.cs#L26). Is this check necessary?